### PR TITLE
Calendar to type script

### PR DESCRIPTION
**Pull request by Inbal-Tish**
Calendar - migrate to ts

### CODE REVIEW

File: src/calendar-list/index.tsx [hunk 1]
```diff
@@ -138,18 +140,19 @@ class CalendarList extends Component<CalendarListProps, CalendarListState> {
     const rows = [];
     const texts = [];
     const date = parseDate(props.current) || new XDate();
+    const {pastScrollRange = 50, futureScrollRange = 50} = props;
```
Comment (ethanshar) [inline src/calendar-list/index.tsx:143]: I see we're setting default prop value for pastScrollRange and futureScrollRange in many places in this file. 
I know it's for typescript cause TS doesn't understand `static defaultProps` anyway let's at least define a const for these default values at the top of the file.

File: src/calendar-list/index.tsx [hunk 2]
```diff
@@ -192,7 +195,7 @@ class CalendarList extends Component<CalendarListProps, CalendarListState> {
   }
 
   scrollToDay(d: XDate, offset: number, animated: boolean) {
-    const {horizontal, calendarHeight, calendarWidth, pastScrollRange, firstDay} = this.props;
+    const {horizontal, calendarHeight = 360, calendarWidth = width, pastScrollRange = 50, firstDay} = this.props;
```
Comment (ethanshar) [inline src/calendar-list/index.tsx:198]: same here, let's declare const values at the top file for better readability..

File: src/calendar-list/index.tsx [hunk 3]
```diff
@@ -348,10 +354,10 @@ class CalendarList extends Component<CalendarListProps, CalendarListState> {
           getItemLayout={this.getItemLayout}
           onViewableItemsChanged={this.onViewableItemsChanged}
           viewabilityConfig={this.viewabilityConfig}
-          initialScrollIndex={this.state.openDate ? this.getMonthIndex(this.state.openDate) : false}
+          initialScrollIndex={this.state.openDate ? this.getMonthIndex(this.state.openDate) : undefined}
           showsVerticalScrollIndicator={showScrollIndicator}
           showsHorizontalScrollIndicator={horizontal && showScrollIndicator}
-          testID={testID}
+          testID={this.props.testID}
```
Comment (ethanshar) [inline src/calendar-list/index.tsx:360]: Why this change? 
why not destructing it from `this.props` as it was..

File: src/calendar-list/item.tsx [hunk 4]
```diff
@@ -101,6 +101,7 @@ class CalendarListItem extends Component<CalendarListItemProps, CalendarListItem
       headerStyle,
       onPressArrowLeft,
       onPressArrowRight,
+      //@ts-ignore
```
Comment (ethanshar) [inline src/calendar-list/item.tsx:104]: Consider using `// @ts-expect-error` if this will be fixed eventually.
Maybe once we'll move other files this will resolve itself and it will notify us about it

File: src/calendar/day/basic/index.tsx [hunk 5]
```diff
@@ -75,13 +75,7 @@ export default class BasicDay extends Component<BasicDayProps> {
   };
 
   get marking() {
-    let marking = this.props.marking || {};
```
Comment (ethanshar) [inline src/calendar/day/basic/index.tsx:78]: It appears this code used to handle a case when marking is an array. 
Are we sure we want to remove it? 
If we're keeping it, there's definitely cleaner way to check for array

File: src/calendar/day/marking/index.tsx [hunk 6]
```diff
@@ -45,14 +50,16 @@ export interface MarkingProps extends DotProps {
   startingDay?: boolean;
   endingDay?: boolean;
   accessibilityLabel?: string;
+  customStyles?: CustomStyle;
```
Comment (ethanshar) [inline src/calendar/day/marking/index.tsx:53]: What is new type? 
I don't we're using it in this file..

File: src/calendar/index.tsx [hunk 7]
```diff
@@ -70,28 +127,21 @@ class Calendar extends Component {
...
+  header: RefObject<CalendarHeader> = React.createRef();
 
-  addMonth = count => {
-    this.updateMonth(this.state.currentMonth.clone().addMonths(count, true));
+  addMonth = (count: number) => {
+    this.updateMonth(this.state.currentMonth.clone().addMonths(count, true), false);
   };
 
-  updateMonth = (day, doNotTriggerListeners) => {
+  updateMonth = (day: any, doNotTriggerListeners: boolean) => {
```
Comment (ethanshar) [inline src/calendar/index.tsx:144]: You can set a default value here so you won't need to pass it on there places 
```suggestion
  updateMonth = (day: any, doNotTriggerListeners?: boolean = false) => {
```

File: src/calendar-list/index.tsx (hunk 3 above)
Comment (Inbal-Tish) [reply src/calendar-list/index.tsx:364]: Just to match other props we're not manipulating before passing

File: src/calendar-list/item.tsx (hunk 4 above)
Comment (Inbal-Tish) [reply src/calendar-list/item.tsx:104]: Done

File: src/calendar-list/index.tsx (hunk 2 above)
Comment (Inbal-Tish) [reply src/calendar-list/index.tsx]: Done

File: src/calendar-list/index.tsx (hunk 1 above)
Comment (Inbal-Tish) [reply src/calendar-list/index.tsx]: Done

File: src/calendar/day/marking/index.tsx (hunk 6 above)
Comment (Inbal-Tish) [reply src/calendar/day/marking/index.tsx:53]: On line #17 I declare the CustomStyle type

File: src/calendar/index.tsx (hunk 7 above)
Comment (Inbal-Tish) [reply src/calendar/index.tsx]: Done

File: src/calendar/day/basic/index.tsx (hunk 5 above)
Comment (Inbal-Tish) [reply src/calendar/day/basic/index.tsx:78]: We're not supporting markings as an Array...

File: src/calendar-list/index.tsx (hunk 3 above)
Comment (ethanshar) [reply src/calendar-list/index.tsx:364]: If anything I would move all `this.props.xxx` to the destructor so it's unified.. no?

**Review by ethanshar — APPROVED** [review]

File: src/calendar-list/index.tsx (hunk 3 above)
Comment (Inbal-Tish) [reply src/calendar-list/index.tsx:364]: I'm hoping to extract FlatList props and pass them as a buldle and not one by one. This is how it's written all over the lib, I was just conforming, it's not a bit issue
